mmv1/products/iamworkforcepool/WorkforcePoolProviderKey.yaml

```diff
@@ -0,0 +1,166 @@
+  api: 'https://cloud.google.com/iam/docs/reference/rest/v1/locations.workforcePools.providers.keys'
+base_url: 'locations/{{location}}/workforcePools/{{workforce_pool_id}}/providers/{{provider_id}}/keys'
+self_link: 'locations/{{location}}/workforcePools/{{workforce_pool_id}}/providers/{{provider_id}}/keys/{{key_id}}'
+create_url: 'locations/{{location}}/workforcePools/{{workforce_pool_id}}/providers/{{provider_id}}/keys?workforcePoolProviderKeyId={{key_id}}'
+immutable: true
+import_format:
+  - 'locations/{{location}}/workforcePools/{{workforce_pool_id}}/providers/{{provider_id}}/keys/{{key_id}}'
+timeouts:
+  insert_minutes: 20
+  delete_minutes: 20
+autogen_async: true
+exclude_sweeper: true
+async:
+  actions: ['create', 'delete']
+  type: 'OpAsync'
+  operation:
+    base_url: '{{op_id}}'
+  result:
+    resource_inside_response: false
+custom_code:
+  constants: 'templates/terraform/constants/iam_workforce_pool_provider_key.go.tmpl'
+examples:
+  - name: 'iam_workforce_pool_provider_saml_key_basic'
+    primary_resource_id: "example"
+    vars:
+      workforce_pool_id: 'example-pool'
+      provider_id: 'example-prvdr'
+      key_id: 'example-key'
+    test_env_vars:
+      org_id: 'ORG_ID'
+parameters:
+properties:
+  - name: 'location'
+    type: String
+    description: |
+      The location for the resource.
+    url_param_only: true
+    required: true
+    immutable: true
+  - name: 'workforcePoolId'
+    type: String
+    description: |
+      The ID of the workforce pool.
+    url_param_only: true
+    required: true
+    immutable: true
+  - name: 'providerId'
+    type: String
+    description: |
+      The ID of the provider.
+    url_param_only: true
+    required: true
+    immutable: true
+  - name: 'keyId'
+    type: String
+    description: |
+      The ID to use for the key, which becomes the final component of the resource name. This value must be 4-32 characters, and may contain the characters [a-z0-9-].
+    url_param_only: true
+    required: true
+    immutable: true
+    validation:
+      function: 'ValidateWorkforcePoolProviderKeyId'
+  - name: 'name'
+    type: String
+    description: |
+      Identifier. The resource name of the key.
+      Format: `locations/{location}/workforcePools/{workforcePoolId}/providers/{providerId}/keys/{keyId}`
+    output: true
+  - name: 'keyData'
+    type: 'NestedObject'
+    description: |
+      Immutable. Public half of the asymmetric key.
+    required: true
+    properties:
+      - name: 'format'
+        type: Enum
+        description: |
+          Output only. The format of the key.
+        output: true
+        enum_values:
+          - 'KEY_FORMAT_UNSPECIFIED'
+          - 'RSA_X509_PEM'
+      - name: 'notBeforeTime'
+        type: String
+        description: |
+          Output only. Earliest timestamp when this key is valid. Attempts to use this key before this time will fail.
```

Comment:
no need to include `output only` in description as it does this automatically when provider is generated. Be sure to address this on all fields. The same applies to `required: true`